Sources/Shared/LocalAuthentication/BiometricsAuthenticationService.swift

```diff
@@ -0,0 +1,142 @@
+import Foundation
+import LocalAuthentication
+import UIKit
+
+public protocol BiometricsAuthenticationServiceProtocol {
+    func checkForBiometrics(controller: UIViewController)
+    func protectAppIfNeeded(controller: UIViewController, completion: (() -> Void)?)
+}
+
+class BiometricsAuthenticationService: BiometricsAuthenticationServiceProtocol {
+    private var context: LAContext?
+
+    private var overlayViewController: BiometricsAuthenticationViewController?
```

Comment:
can we talk to this without knowing it's a UIViewController?

Reply:
I made it agnostic to controller class now, but still a UIViewController reference. Are you concerned about testability?

Reply:
Done, now it replaces the rootViewController of the window with the lock screen and then replaces back after authenticating